dvc/remote/base.py

```diff
@@ -820,9 +826,102 @@ def cache_exists(self, checksums, jobs=None, name=None):
         Returns:
             A list with checksums that were found in the remote
         """
-        if not self.no_traverse:
-            return list(set(checksums) & set(self.all()))
 
+        if self.no_traverse:
+            return self._cache_exists_no_traverse(checksums, jobs, name)
+
+        # Fetch one parent cache dir for estimating size of entire remote cache
+        checksums = frozenset(checksums)
+        prefix = "0" * self.TRAVERSE_PREFIX_LEN
+        traverse_dirs = pow(16, self.TRAVERSE_PREFIX_LEN)
+        remote_checksums = set(
+            map(self.path_to_checksum, self.list_cache_paths(prefix=prefix))
+        )
+        if not remote_checksums:
+            remote_size = traverse_dirs
+        else:
+            remote_size = traverse_dirs * len(remote_checksums)
+        logger.debug("Estimated remote size: {} files".format(remote_size))
```

Comment:
I really like how loggers are used here 👍 we should be doing this more everywhere ... I miss this stuff a lot in DVC - it make most of our logs "unusable" and we have to ask people questions - a lot of questions